### PR TITLE
symitriDapRtdProvider : Hash user identity before using it

### PR DESCRIPTION
**Pull request by mp4symitri**
## Type of change

- [x] Feature

## Description of change

Added custom sha256 function to synchronously hash user identity as asynchronous hashing function can't be called from synchronous code.
Added code to ensure user identity is hashed before using it.
Updated test cases to test user identifier and hashing algorithm.

### CODE REVIEW

File: modules/symitriDapRtdProvider.js [hunk 1]
```diff
@@ -526,6 +525,125 @@ export function createRtdProvider(moduleName, moduleCode, headerPrefix) {
...
+     * Copyright 2022 Andrea Griffini
+     * https://github.com/6502/sha256/
+     *
+     * PARAMETERS
+     *
+     * message: any string that needs tobe hashed. In this case it will be user identifier
+     *
+     */
+
+    generateHash: function(message) {
```
[the remark this answers is not in the document]
Comment (mp4symitri) [reply modules/symitriDapRtdProvider.js:556]: @patmmccann, Browser's built in function uses SubtleCrypto interface that provides crypto.subtle.digest to generate digest. However this digest function is an asynchronous function whereas we want to generate this hash from a synchronous code. 
We do use SubtleCrypto for calculating entropy asynchronously but here we needed a synchronous version.

[the remark this answers is not in the document]
Comment (jpalladino84) [reply modules/symitriDapRtdProvider.js:556]: @patmmccann should this be called from a async call instead? 

Are there any other concerns that we need to consider?

File: modules/symitriDapRtdProvider.md [hunk 2]
```diff
@@ -1,49 +1,88 @@
...
+title: Symitri DAP Real Time Data Provider Module
+display_name: Symitri DAP Real Time Data Provider Module
+description: Symitri DAP Real Time Data Provider Module
+page_type: module
+module_type: rtd
+module_code : symitriDapRtdProvider
+enable_download : true
+vendor_specific: true
+sidebarType : 1
+---
```
Comment (ChrisHuie) [inline modules/symitriDapRtdProvider.md:12]: This should be in our docs repo as it is labels for our static site generator -> https://github.com/prebid/prebid.github.io. The markdown file here does not connect at all to our docs site and just lives in the PBjs project as reference

**Comment by jpalladino84** [comment]
Hi @ncolletti 

Thanks to @ChrisHuie the build is passing and I update the documentation on [prebid.github.io ](https://github.com/prebid/prebid.github.io/pull/5569).

Let me know if you have any questions or concerns!

**Review by ChrisHuie — APPROVED** [review]

**Pull request merged**